yt_dlp/extractor/opencast.py

```diff
@@ -0,0 +1,230 @@
+                            oc-presentation\.ltcc\.tuwien\.ac\.at|
+                            medien\.ph-noe\.ac\.at|
+                            oc-video\.ruhr-uni-bochum\.de|
+                            oc-video1\.ruhr-uni-bochum\.de|
+                            opencast\.informatik\.uni-goettingen\.de|
+                            heicast\.uni-heidelberg\.de|
+                            opencast\.hawk\.de:8080|
+                            opencast\.hs-osnabrueck\.de|
+                            video[0-9]+\.virtuos\.uni-osnabrueck\.de|
+                            opencast\.uni-koeln\.de|
+                            media\.opencast\.hochschule-rhein-waal\.de|
+                            matterhorn\.dce\.harvard\.edu|
+                            hs-harz\.opencast\.uni-halle\.de|
+                            videocampus\.urz\.uni-leipzig\.de|
+                            media\.uct\.ac\.za|
+                            vid\.igb\.illinois\.edu|
+                            cursosabertos\.c3sl\.ufpr\.br|
+                            mcmedia\.missioncollege\.org|
+                            clases\.odon\.edu\.uy
+                        )'''
+    _UUID_RE = r'[\da-fA-F]{8}-[\da-fA-F]{4}-[\da-fA-F]{4}-[\da-fA-F]{4}-[\da-fA-F]{12}'
+
+    def _call_api(self, host, video_id, note=None, errnote=None, fatal=True):
+        return self._download_json(self._API_BASE % (host, video_id), video_id, note=note, errnote=errnote, fatal=fatal)
+
+    def _parse_mediapackage(self, video):
+        tracks = try_get(video, lambda x: x['media']['track'])
+
+        video_id = video.get('id')
+
+        formats = []
+        for track in tracks:
+            href = track['url']
+            ext = determine_ext(href, None)
+            track_obj = {'url': href}
+
+            transport = track.get('transport')
+
+            if transport == 'DASH' or ext == 'mpd':
+                formats.extend(self._extract_mpd_formats_and_subtitles(href, video_id, mpd_id='dash', fatal=False))
+            elif transport == 'HLS' or ext == 'm3u8':
+                formats.extend(
+                    self._extract_m3u8_formats_and_subtitles(href, video_id, m3u8_id='hls', entry_protocol='m3u8_native', fatal=False)
+                )
+            elif transport == 'HDS' or ext == 'f4m':
+                formats.extend(self._extract_f4m_formats(href, video_id, f4m_id='hds', fatal=False))
+            elif transport == 'SMOOTH':
+                formats.extend(self._extract_ism_formats(href, video_id, ism_id='smooth', fatal=False))
+            elif ext == 'smil':
+                formats.extend(self._extract_smil_formats(href, video_id, fatal=False))
+            else:
+                if transport is not None:
+                    track_obj.update({'format_note': track.get('transport')})
+                    if transport == 'RTMP':
+                        m_obj = re.search(r'^(?:rtmp://[^/]+/(?P<app>[^/]+))/(?P<playpath>.+)$', href)
```

Comment:
```suggestion
                        m_obj = re.match(r'(?:rtmp://[^/]+/(?P<app>[^/]+))/(?P<playpath>.+)', href)
```

also I'd prefer to add a group for extension extraction (I won't force you unless the maintainer says the same)